### PR TITLE
feat(j-s): Show district court abbreviation on mobile 

### PR DESCRIPTION
**Pull request by oddsson**
# Show district court abbreviation for public prosecutor

[Asana](https://app.asana.com/0/1199153462262248/1208639564800956/f)

## What

In the case list for public prosecutor, show a abbreviation of the district court's name with the court case number.

## Why

The staff at the public prosecutors need to see which district court sent a case

## Screenshots / Gifs

Screenshot 2024-11-28 at 13 41 15

## Checklist:

- [x] I have performed a self-review of my own code
- [ ] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [x] Formatting passes locally with my changes
- [x] I have rebased against main before asking for a review

## Release Notes

- **New Features**
  - Added a `court` property to case entries, enhancing data related to court associations.
  - Introduced `districtCourtAbbreviation` function for improved display of court case numbers.
  - Enhanced GraphQL query for cases to include court information.

- **Improvements**
  - Table components now allow sorting by case number and other relevant fields for better user interaction.
  - Improved rendering logic for displaying court case numbers with abbreviations for clarity.

### CODE REVIEW

File: apps/judicial-system/web/src/components/Table/Table.tsx [hunk 1]
```diff
@@ -168,25 +171,30 @@ const Table: FC<TableProps> = (props) => {
...
-          }
-
-          return entry[sortConfig.column]?.toString()
-        }
-
         const compareResult = compareLocaleIS(
-          getColumnValue(a),
-          getColumnValue(b),
+          getColumnValue(a, sortConfig.column),
+          getColumnValue(b, sortConfig.column),
```
[the remark this answers is not in the document]
Comment (oddsson) [reply apps/judicial-system/web/src/components/Table/Table.tsx:197]: Good catch! Thanks

File: apps/judicial-system/web/src/routes/Shared/Cases/cases.graphql [hunk 2]
```diff
@@ -34,6 +34,10 @@ query Cases {
     initialRulingDate
     rulingDate
     rulingSignatureDate
+    court {
```
Comment (gudjong) [inline apps/judicial-system/web/src/routes/Shared/Cases/cases.graphql:37]: Is this also needed in other case list queries, such as for the prison admin or defenders?

**Review by gudjong — APPROVED** [review]
Very nice. Just one question.

File: apps/judicial-system/web/src/routes/Shared/Cases/cases.graphql (hunk 2 above)
Comment (oddsson) [reply apps/judicial-system/web/src/routes/Shared/Cases/cases.graphql:37]: Not currently. @thorhildurt is however working on adding this feature for prisons so this needs to be addressed there.

File: apps/judicial-system/web/src/routes/Shared/Cases/MobileCase.tsx [hunk 3]
```diff
@@ -91,7 +93,12 @@ const MobileCase: FC<PropsWithChildren<Props>> = ({
       <Text title={theCase.policeCaseNumbers?.join(', ')}>
         {displayFirstPlusRemaining(theCase.policeCaseNumbers)}
       </Text>
-      {theCase.courtCaseNumber && <Text>{theCase.courtCaseNumber}</Text>}
+
+      {theCase.courtCaseNumber && (
+        <Text>{`${courtAbbreviation ? `${courtAbbreviation}: ` : ''}${
+          theCase.courtCaseNumber
+        }`}</Text>
+      )}
```
[the remark this answers is not in the document]
Comment (oddsson) [reply apps/judicial-system/web/src/routes/Shared/Cases/MobileCase.tsx:101]: No I think it's fine

**Pull request merged**